gammapy/spectrum/fit.py

```diff
@@ -254,18 +254,18 @@ def _calc_statval_helper(self, obs, prediction):
         """
         # Off stat = 0 by default
         off_stat = np.zeros(obs.e_reco.nbins)
-        if self.stat == 'cash':
+        if self.stat == 'cash' or self.stat == 'cstat':
             if self.background_model is not None:
                 mu_on = prediction[0] + prediction[1]
-                on_stat = stats.cash(n_on=obs.on_vector.data.data.value,
-                                     mu_on=mu_on)
+                on_stat = stats.__dict__[self.stat](n_on=obs.on_vector.data.data.value,
```

Comment:
This is not so nice. I would suggest to at least not do this three times, and instead do this once at the top of the function:
```python
stats_func = getattr(stats, self.stat)
```
and then call `stats_func` on the three lines.